### PR TITLE
Automattic Components: Temporarily shim old calypso form label component

### PR DESCRIPTION
**Pull request by jeyip**
Related to https://github.com/Automattic/wp-calypso/pull/79256
Tracking issue https://github.com/Automattic/martech/issues/2023

## Proposed Changes
- At dale's suggestion, point the old form label component to the new one in @automattic/components, allowing us to transition from the old code to the new in smaller pieces and without risking updates to the old component in the interim.
- Remove old files

## Testing Instructions

* Check out this branch or use calypso.live
* VideoPress Setup
  * Navigate to calypso.localhost:3000/setup/videopress/options
  * Screenshot 2023-08-09 at 2 44 14 PM
* /me < Privacy
  * Navigate to calypso.localhost:3000/me/security/two-step
  * Click on "Register key"
  * Screenshot 2023-08-09 at 2 53 18 PM
  * Click on "Add new application password"
  * Screenshot 2023-08-09 at 2 50 24 PM
* Domain Management
  * Navigate to calypso.localhost:3000/domains/manage/all/{SITE_SLUG_FOR_CUSTOM_DOMAIN}/add-dns-record/{SITE_SLUG_FOR_CUSTOM_DOMAIN}
  * Screenshot 2023-08-09 at 2 59 20 PM
* Importer
  * Navigate to calypso.localhost:3000/migrate/{SITE_SLUG}
  * Click on import from WordPress
  * Screenshot 2023-08-09 at 3 03 58 PM
* Hosting Configuration
  * Navigate to a site with a business plan calypso.localhost:3000/hosting-config/{SITE_SLUG}
  * Screenshot 2023-08-09 at 3 06 06 PM
* Smoke test other instances of form label to ensure that they are rendered properly

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by jeyip** [comment]
@chriskmnds + @jdc91 this PR is ready for review 😎

File: client/components/domains/registrant-extra-info/test/uk-form.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ import { render, screen } from '@testing-library/react';
 import { RegistrantExtraInfoUkForm } from '../uk-form';
 
 jest.mock( '@automattic/components', () => ( {
+	FormLabel: ( { children } ) => (
+		<div data-testid="test-label" key={ children }>
+			{ children }
+		</div>
+	),
```
Comment (ddc22) [inline client/components/domains/registrant-extra-info/test/uk-form.js]: Have you tried this using [requireActual](https://jestjs.io/docs/jest-object#jestrequireactualmodulename)?
You can spread the module over  the mock and ONLY the mocked attribute can be overridden.

File: client/components/forms/form-label/index.tsx [hunk 2]
```diff
@@ -1,38 +1,8 @@
...
-			{ hasChildren && optional && (
-				<small className="form-label__optional">{ translate( 'Optional' ) }</small>
-			) }
-		</label>
-	);
-};
+// We're in the process of migrating to @automattic/components. Because of this, we're using this wrapper
+// component to point references to the old count component to the new one in @automattic/components.
+// This allows us to transition in smaller pieces and without risking updates to the old Calypso component in the interim.
+// See https://github.com/Automattic/martech/issues/2028 for more details.
```
Comment (ddc22) [inline client/components/forms/form-label/index.tsx:6]: Nice !

**Review by ddc22 — APPROVED** [review]
Change works as expected! I have left a comment to be addressed below related tests.
Can be 🚢 ed after resolving 🥳

File: client/components/domains/registrant-extra-info/test/uk-form.js (hunk 1 above)
Comment (ddc22) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: Eg:
```javascript
jest.mock('./someModule', () => ({
  ...jest.requireActual('./someModule'),
  api: () => Promise.resolve('foo'),
}));
```

Comment (jeyip) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: I did, but it didn't appear to work for some reason. 🤔 

Let me try again.

Comment (jeyip) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: `TypeError: Cannot read properties of undefined (reading 'default')`

For some reason, it seems like elements of @automattic/component are unavailable in the `uk-form.js` test environment. I'll dig a little bit and timebox the investigation

Comment (ddc22) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: I guess you can merge this and then explore if you're curious 👍 Totally up-to you, I am ok either way.

Comment (jeyip) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: So odd. There's something about the way packages are resolved in jest.

```javascript
// This works
`...jest.requireActual( '../../../../../packages/components' ).default`

// This doesn't work
`...jest.requireActual( '@automattic/components' )`
```

It also seems like @automattic/components isn't even available in this test environment for some reason

Comment (jeyip) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: > I guess you can merge this and then explore if you're curious 👍 Totally up-to you, I am ok either way.

Thanks Dehan! I'll spend a little bit longer, but will make a follow-up if it doesn't turn up anything fruitful

Comment (jeyip) [reply client/components/domains/registrant-extra-info/test/uk-form.js]: @dehan I ended up modifying the test to avoid using any mocks in the file 😅 The more I looked at the specs, the more I realized that they were unnecessary.

Instead, I set the expectations up to run against the actual @automattic/component implementations [05feb71](https://github.com/Automattic/wp-calypso/pull/80350/commits/05feb718697ceaa7b83676c78d644085aaaa87ad)
